### PR TITLE
[Builtins] Add TextViewer Builtin

### PR DESCRIPTION
**Pull request by phil65**
This adds a builtin to open the TextViewer dialog filled with header / text defined by params.
Can be useful for example to show plot inside a separate dialog (FireTV does that too and it works great)
At the moment skinners would have to create separate xml dialogs for this or have to use hacks, or my ToolBox script. It´s better to have this in core though I think, that way Python devs also have quick access to such functionality (at the moment they have to create a window class, load the xml dialog and fill the labels manually)

### CODE REVIEW

**Comment by phil65** [comment]
@mkortstiege , @xhaggi for review

**Comment by mkortstiege** [comment]
Looks good to me.

File: xbmc/interfaces/Builtins.cpp [hunk 1]
```diff
@@ -167,6 +168,7 @@ const BUILT_IN commands[] = {
   { "CancelAlarm",                true,   "Cancels an alarm" },
   { "Action",                     true,   "Executes an action for the active window (same as in keymap)" },
   { "Notification",               true,   "Shows a notification on screen, specify header, then message, and optionally time in milliseconds and a icon." },
+  { "textviewer",                 true,   "Opens TextViewer Dialog, specify header, then message." },
```
Comment (xhaggi) [inline xbmc/interfaces/Builtins.cpp]: minor but please use camel case like all other

File: xbmc/interfaces/Builtins.cpp [hunk 2]
```diff
@@ -1226,6 +1228,15 @@ int CBuiltins::Execute(const std::string& execString)
     else
       CGUIDialogKaiToast::QueueNotification(params[0],params[1]);
   }
+  else if (execute == "textviewer")
+  {
+    if (params.size() < 2)
+      return -1;
+    CGUIDialogTextViewer* pDlgInfo = (CGUIDialogTextViewer*)g_windowManager.GetWindow(WINDOW_DIALOG_TEXT_VIEWER);
+    pDlgInfo->SetHeading(params[0]);
```
Comment (xhaggi) [inline xbmc/interfaces/Builtins.cpp]: null check before you use the dialog

**Comment by xhaggi** [comment]
it would be nice if you drop the indentation changes or separate them into another commit

File: xbmc/interfaces/Builtins.cpp (hunk 1 above)
Comment (uNiversaI) [inline xbmc/interfaces/Builtins.cpp]: { "textviewer",                 true,   "Opens a textviewer dialog, specify header, then message." },

Comment (xhaggi) [inline xbmc/interfaces/Builtins.cpp]: btw we should use "OpenTextViewer" which makes it more clear what this cmd will do.

**Comment by ronie** [comment]
nice one. i've indeed been using a custom dialog for movie plots, album reviews and artist descriptions.
so this would be very welcome.

as for the action name, we usually use short commands, like 'slideshow' or 'notification'
instead of showfoo or openbar. i'm fine either way though.

**Comment by BigNoid** [comment]
While testing this with 
```<onclick>TextViewer($INFO[ListItem.Label],$INFO[ListItem.Plot])</onclick> ```
in Confluence, the plot is always cutoff after the first comma in the actual plot.

**Comment by ronie** [comment]
$ESCINFO[] ?

**Comment by BigNoid** [comment]
Ah that's right, sorry for noise :)

File: xbmc/interfaces/Builtins.cpp (hunk 2 above)
Comment (phil65) [inline xbmc/interfaces/Builtins.cpp]: you mean something like 

if (pDlgInfo == NULL)
     return;

? (I´m no C programmer).
There is no null check for other similar calls too as far as I can tell.

Comment (xhaggi) [inline xbmc/interfaces/Builtins.cpp]:
```
if (pDlgInfo) 
{
  pDlgInfo->SetHeading(params[0]);
  ...
}
```

should the way you go

Comment (phil65) [inline xbmc/interfaces/Builtins.cpp]: ok, thx for reviewing.

File: xbmc/interfaces/Builtins.cpp (hunk 1 above)
Comment (phil65) [inline xbmc/interfaces/Builtins.cpp]: will CamelCase it, "TextViewer" is bettter though than "Open..." or "Show ..."

Comment (xhaggi) [inline xbmc/interfaces/Builtins.cpp]: okay

**Comment by phil65** [comment]
Updated and split into two separate commits.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by phil65** [comment]
@MartijnKaijser this shouldnt be merged (see discussion in 	https://github.com/xbmc/xbmc/pull/7217 ) I will close this one, the other one can stay open then as a reminder.)

**Comment by MartijnKaijser** [comment]
ah i thought this was ok any way.
don't count on the issue being resolved any time soon

**Pull request closed without merging**